src/Interfaces/AuthenticationContainerInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+declare(strict_types=1);
+
+namespace Firehed\API\Interfaces;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * SHOULD use FQCN for get/has - etc
+ */
+interface AuthenticationContainerInterface extends ContainerInterface
+{
+    public function isAuthenticated(): bool;
```

Comment:
This may be too simple; instead, it may be better to just kick all of the logic into the container values.

Given that, it would be possible to drop this interface entirely and rename the consumers, but the additional semantic value from the name might be worthwhile.

See also: AuthenticationProvider comment